src/Jobs/Notifications/AbstractItemNotificationJob.php

```diff
@@ -35,9 +36,17 @@ protected function process_items( array $args ): void {
 		$topic = $args['topic'];
 		$data  = $args['data'] ?? [];
 
-		if ( $this->can_process( $item, $topic ) && $this->notifications_service->notify( $topic, $item, $data ) ) {
-			$this->set_status( $item, $this->get_after_notification_status( $topic ) );
-			$this->handle_notified( $topic, $item );
+		try {
+			if ( $this->can_process( $item, $topic ) && $this->notifications_service->notify( $topic, $item, $data ) ) {
+				$this->set_status( $item, $this->get_after_notification_status( $topic ) );
+				$this->handle_notified( $topic, $item );
+			}
+		} catch ( InvalidValue $e ) {
+			do_action(
+				'woocommerce_gla_error',
+				sprintf( 'Error sending Notification for - Item ID: %s - Topic: %s - Data %s. Product was deleted from the database before the notification was sent.', $item, $topic, json_encode( $data ) ),
+				__METHOD__
+			);
```

Comment:
This handles the potential case in which we attempt to get an already deleted item from the database.